src/components/Select/MultiSelect.tsx

```diff
@@ -0,0 +1,117 @@
+import { forwardRef, useCallback, useEffect, useState } from "react";
+
+import SelectContainer from "./common/SelectContainer";
+import { SelectContainerProps } from "./types";
+import SelectTrigger from "./common/SelectTrigger";
+import { SelectContent, SelectContentProps } from "./common/SelectContent";
+import { SelectGroup, SelectItem, SelectNoData } from "./common/SelectContenOptions";
+
+interface Props {
+  defaultValue?: Array<string>;
+  onChange?: (value: Array<string>) => void;
+  value?: Array<string>;
+  defaultOpen?: boolean;
+  open?: boolean;
+  onOpenChange?: (open: boolean) => void;
+}
+
+export type MultiSelectProps = Omit<
+  SelectContainerProps,
+  "onChange" | "value" | "open" | "onOpenChange" | "onSelect"
+> &
+  Props;
+
+export const MultiSelect = ({
+  children,
+  value: valueProp,
+  defaultValue,
+  onChange: onChangeProp,
+  open: openProp,
+  defaultOpen,
+  onOpenChange: onOpenChangeProp,
+  ...props
+}: MultiSelectProps) => {
+  const [selectedValues, setSelectedValues] = useState<Array<string>>(
+    valueProp ?? defaultValue ?? []
+  );
+  const [open, setOpen] = useState(openProp ?? defaultOpen ?? false);
+  const onOpenChange = useCallback(
```

Comment:
the open state should be totally controlled by the user or uncontrolled. If the user passes the `open` prop than we should always respect that without assuming control of the element

Reply:
same thing should apply to the `value` prop

Reply:
Added tests for the open prop will add for the value too